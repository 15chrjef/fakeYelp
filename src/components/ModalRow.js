import React from 'react';

const ModalRow = (props) => {
	const { label, name, id, onChange } = props
	let tag;
	if(id === 'description'){
		tag = (
			<textarea 
				style={{width: '70%', height: '150px', maxWidth: '70%', maxHeight: '150px', marginLeft: '60px'}}
				type='text' 
				id={id} 
				value={name} 
				onChange={onChange}
			/>
		)
	}else {
		tag =(	
			<input 
				style={{width: '70%',  maxWidth: '70%'}}
				type='text' 
				id={id} 
				value={name} 
				onChange={onChange}
			/>
		)
	}
	return(
		<div style={{display: 'flex', justifyContent: 'space-between', width: '90%', marginTop: '10px'}}>
			<label style={{fontWeight: 'bold'}}>{label}</label>
			{tag}
		</div>
	)
}

export default ModalRow;