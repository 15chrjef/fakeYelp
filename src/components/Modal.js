import React from 'react';
import ModalRow from './ModalRow'
export default class Modal extends React.Component {
	constructor(props){
		super(props)
		this.state = {
			name: '',
			description: '',
			stars : '',
			subcategories: '',
			price: '',
			bKey: ''
		}
	}
	onChange(e) {
		switch(e.target.id){
			case 'name':
				this.setState({ name: e.target.value})
				break
			case 'description':
				this.setState({ description: e.target.value})
				break
			case 'stars':
				this.setState({ stars: e.target.value})
				break
			case 'categories':
				this.setState({ subcategories: e.target.value})
				break
			case 'price':
				this.setState({ price: e.target.value})
			default:
		}
	}
	componentWillReceiveProps(newProps){
		console.log('mounting mounting', newProps)
		const { bKey } = this.state;
		if(newProps.editingBusiness !== '' && newProps.editingBusiness !== null){
			const { name, description, reviews, subcategories, price, bKey } = newProps.editingBusiness
			console.log('businesssss', this.state)
			this.setState({
				name,
				description,
				stars: reviews[0].rating,
				subcategories,
				price,
				bKey
			})
		}
	}
	handleSubmit(){
		this.props.createOrUpdateBusiness(this.state)
		this.props.displayModal()
		this.setState({
			name: '',
			description: '',
			stars : '',
			subcategories: '',
			price: '',
		})
	}
	render(){
		const { name, description, stars, categories, price } = this.state
		const title = this.props.type === 'create' ? 'Create a Business': 'Update the Description';
			return(
				<div className='modal' style={this.props.show}>
					<h3 style={{color: 'red'}}>{title}</h3>
					<div className='container'>
						<ModalRow id='name' value={name} onChange={this.onChange.bind(this)} label='Name'/>
						<ModalRow id='description' value={description} onChange={this.onChange.bind(this)} label='Description'/>
						<ModalRow id='stars' value={stars} onChange={this.onChange.bind(this)} label='Stars'/>
						<ModalRow id='categories' value={categories} onChange={this.onChange.bind(this)} label='Categories'/>
						<ModalRow id='price' value={price} onChange={this.onChange.bind(this)} label='Price'/>
					</div>
					<div className='container'>				
						<input onClick={this.handleSubmit.bind(this)} style={{marginTop: '10px'}} type='submit'/>
						<i 
							onClick={this.props.displayModal} 
							style={{color: 'red', alignSelf: 'right', cursor: 'pointer', marginTop: '12px'}} 
							className="fa fa-minus-circle fa-2x">
						</i>
					</div>
				</div>
			)
	}
}
