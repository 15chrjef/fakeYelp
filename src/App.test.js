import React from 'react';
import renderer from 'react-test-renderer';
import ModalRow from './components/ModalRow'
import Modal from './components/Modal'

it('renders Modal and ModalRow without crashing', () => {
  const row = renderer.create(
		<ModalRow id='name' value='' label='Name' />
	);
  let modalRowTree = row.toJSON();
	expect(modalRowTree).toMatchSnapshot();

  const modal = renderer.create(
		<Modal />
	);
  let modalTree = modal.toJSON();
	expect(modalTree).toMatchSnapshot();
});
