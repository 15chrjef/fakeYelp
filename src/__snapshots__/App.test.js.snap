// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Modal and ModalRow without crashing 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "marginTop": "10px",
      "width": "90%",
    }
  }
>
  <label
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Name
  </label>
  <input
    id="name"
    onChange={undefined}
    style={
      Object {
        "maxWidth": "70%",
        "width": "70%",
      }
    }
    type="text"
    value={undefined}
  />
</div>
`;

exports[`renders Modal and ModalRow without crashing 2`] = `
<div
  className="modal"
  style={undefined}
>
  <h3
    style={
      Object {
        "color": "red",
      }
    }
  >
    Update the Description
  </h3>
  <h5
    style={
      Object {
        "color": "red",
      }
    }
  >
    Must have all fields filled for Creation
  </h5>
  <div
    className="container"
  >
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
          "width": "90%",
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Name
      </label>
      <input
        id="name"
        onChange={[Function]}
        style={
          Object {
            "maxWidth": "70%",
            "width": "70%",
          }
        }
        type="text"
        value={undefined}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
          "width": "90%",
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Description
      </label>
      <textarea
        id="description"
        onChange={[Function]}
        style={
          Object {
            "height": "150px",
            "marginLeft": "60px",
            "maxHeight": "150px",
            "maxWidth": "70%",
            "width": "70%",
          }
        }
        type="text"
        value={undefined}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
          "width": "90%",
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Stars
      </label>
      <input
        id="stars"
        onChange={[Function]}
        style={
          Object {
            "maxWidth": "70%",
            "width": "70%",
          }
        }
        type="text"
        value={undefined}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
          "width": "90%",
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Categories
      </label>
      <input
        id="categories"
        onChange={[Function]}
        style={
          Object {
            "maxWidth": "70%",
            "width": "70%",
          }
        }
        type="text"
        value={undefined}
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginTop": "10px",
          "width": "90%",
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Price
      </label>
      <input
        id="price"
        onChange={[Function]}
        style={
          Object {
            "maxWidth": "70%",
            "width": "70%",
          }
        }
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <div
    className="container"
  >
    <input
      onClick={[Function]}
      style={
        Object {
          "marginTop": "10px",
        }
      }
      type="submit"
    />
    <i
      className="fa fa-minus-circle fa-2x"
      onClick={undefined}
      style={
        Object {
          "alignSelf": "right",
          "color": "red",
          "cursor": "pointer",
          "marginTop": "12px",
        }
      }
    />
  </div>
</div>
`;
